source/common/http/async_client_impl.h

```diff
@@ -261,6 +261,19 @@ class AsyncStreamImpl : public AsyncClient::Stream,
   void continueDecoding() override { NOT_IMPLEMENTED; }
   void addDecodedData(Buffer::Instance&, bool) override { NOT_IMPLEMENTED; }
   const Buffer::Instance* decodingBuffer() override { return buffered_body_.get(); }
+  void sendLocalReply(Code code, const std::string& body,
+                      std::function<void(HeaderMap& headers)> modify_headers) override {
+    Utility::sendLocalReply(
```

Comment:
Not sure if this is correct for an HTTP client; do we ever hit this in the configured client filter stack? I.e. could it be `NOT_IMPLEMENTED`?

Reply:
I initially left this empty, and when tests failed put NOT_REACHED and it was reached. With this it works, and I did not dig deeper.

Reply:
Yeah we can definitely hit this for AsyncClient, since it goes through router and can return 503 for no healthy upstream and a bunch of other reasons.

Reply:
Right, that makes sense.